server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/parquet/ParquetTypeConverter.java

```diff
@@ -219,28 +235,86 @@ public Object getValue(Group group, int columnIndex, int repeatIndex, Type type)
         public void addValueToJsonArray(Group group, int columnIndex, int repeatIndex, Type type, ArrayNode jsonNode) {
             jsonNode.add(group.getBoolean(columnIndex, repeatIndex));
         }
-    };
-
-
-    public static ParquetTypeConverter from(PrimitiveType primitiveType) {
-        return valueOf(primitiveType.getPrimitiveTypeName().name());
-    }
-
+    },
 
-    // ********** PUBLIC INTERFACE **********
-    public abstract DataType getDataType(Type type);
+    LIST {
+        @Override
+        public DataType getDataType(Type type) {
+            Type elementType = getElementType(type.asGroupType());
+            if (!elementType.isPrimitive()) {
+                String originalTypeName = elementType.asGroupType().getOriginalType() == null ?
+                        "non primitives" :
+                        elementType.asGroupType().getOriginalType().name();
+                throw new UnsupportedTypeException(String.format("Parquet LIST of %s is not supported.", originalTypeName));
+            }
+            return from(elementType).getArrayDataType(elementType);
+        }
 
-    public abstract Object getValue(Group group, int columnIndex, int repeatIndex, Type type);
+        @Override
+        public Object getValue(Group group, int columnIndex, int repeatIndex, Type type) {
+            PgArrayBuilder pgArrayBuilder = new PgArrayBuilder(pgUtilities);
+            pgArrayBuilder.startArray();
+
+            Group listGroup = group.getGroup(columnIndex, repeatIndex);
+            // a listGroup can have any number of repeatedGroups
+            int repetitionCount = listGroup.getFieldRepetitionCount(0);
+            boolean elementNeedsEscapingInArray = getDataType(type).getTypeElem().getNeedsEscapingInArray();
+            PrimitiveType elementType = getElementType(type.asGroupType()).asPrimitiveType();
+            ParquetTypeConverter converter = from(elementType);
```

Comment:
it is a bit easier to reason about this logic if I rewrite it as such (avoiding multiple lookups and deriving it from elementType):
```suggestion
            PrimitiveType elementType = getElementType(type.asGroupType()).asPrimitiveType();
            ParquetTypeConverter elementConverter = from(elementType);
            boolean elementNeedsEscapingInArray = elementConverter.getDataType(elementType).getNeedsEscapingInArray();
```
Will this still be equivalent and correct ?